### PR TITLE
docs(dev): fix typos

### PR DESCRIPTION
**Pull request by aikooo7**
# Changes

listitems -> list items
random places-that -> random places that

### CODE REVIEW

**Comment by aikooo7** [comment]
My bad if the commit message is wrong, now re-seeing it I figured out I should've put "fix typos" and not "fixed typos" but I guess I can't change it?

**Comment by aikooo7** [comment]
By the message in the failed test, I don't think that is part of my commit:

```
FAILED   1 test, listed below:
FAILED   test/functional/plugin/lsp_spec.lua @ 278: LSP basic_init test should detach buffer in response to nvim_buf_detach
test/functional/plugin/lsp_spec.lua:301: exit code
Expected objects to be the same.
Passed in:
(number) 101
Expected:
(number) 0

stack traceback:
	(tail call): ?
	test/functional/plugin/lsp_spec.lua:301: in function 'on_exit'
	test/functional/plugin/lsp/helpers.lua:167: in function 'test_rpc_server'
	test/functional/plugin/lsp_spec.lua:284: in function <test/functional/plugin/lsp_spec.lua:278>


 34 SKIPPED TESTS
 1 FAILED TEST
------------------------------------------------------------------------------
$NVIM_LOG_FILE: /home/runner/work/neovim/neovim/build/.nvimlog
(last 100 lines)
ERR 2023-11-11T22:48:59.021 T1342.9963.0 libuv_process_spawn:97: uv_spawn(/usr/bin/pwd) failed: permission denied
ERR 2023-11-11T22:49:04.095 T1384.10171.0 libuv_process_spawn:97: uv_spawn(nosuchshell) failed: no such file or directory
ERR 2023-11-11T22:49:04.096 T1384.10171.0 libuv_process_spawn:97: uv_spawn(nosuchshell) failed: no such file or directory
ERR 2023-11-11T22:49:07.255 T1446.10480.0/c os_set_cloexec:488: Failed to get flags on descriptor 3: Bad file descriptor
WRN 2023-11-11T22:51:10.880 T3638.14779.0/c server_start:164: Failed to start server: no such file or directory: /home/runner/work/neovim/neovim/build/Xtest_tmpdir/a%b.runner/9jNwjV/a/
WRN 2023-11-11T22:51:10.889 T3638.14779.0/c server_start:164: Failed to start server: no such file or directory: /home/runner/work/neovim/neovim/build/Xtest_tmpdir/a%b%c.runner/iphRVe/a/
ERR 2023-11-11T22:52:22.048 nvim.18123.0/c chan_close_with_error:647: RPC: (null)
ERR 2023-11-11T22:52:33.603 nvim.19060.0 tui_mode_change:1163: uv_tty_set_mode failed: inappropriate ioctl for device
ERR 2023-11-11T22:52:46.795 T4958.19545.0 decor_provider_error:32: error in provider ns1.start: Error executing lua: [string "<nvim>"]:4: Foo
stack traceback:
	[C]: in function 'error'
	[string "<nvim>"]:4: in function <[string "<nvim>"]:3>
ERR 2023-11-11T22:52:57.210 T5094.19865.0 os_set_cloexec:488: Failed to get flags on descriptor 3: Bad file descriptor
ERR 2023-11-11T22:54:28.943 T6651.23348.0 libuv_process_spawn:97: uv_spawn(fake_shell) failed: no such file or directory
------------------------------------------------------------------------------
-- Tests exited non-zero: 1
CMake Error at /home/runner/work/neovim/neovim/cmake/RunTests.cmake:107 (message):
  functional tests failed with error: 1


-- Output to stderr:
E5113: Error while calling lua chunk: 

FAILED: test/CMakeFiles/functionaltest /home/runner/work/neovim/neovim/build/test/CMakeFiles/functionaltest 
cd /home/runner/work/neovim/neovim/build/test && /usr/local/bin/cmake -D NVIM_PRG=/home/runner/work/neovim/neovim/build/bin/nvim -D WORKING_DIR=/home/runner/work/neovim/neovim -D BUSTED_OUTPUT_TYPE=nvim -D TEST_DIR=/home/runner/work/neovim/neovim/test -D BUILD_DIR=/home/runner/work/neovim/neovim/build -D DEPS_INSTALL_DIR=/home/runner/work/neovim/neovim/build/usr -D TEST_TYPE=functional -D CIRRUS_CI= -D CI_BUILD=ON -P /home/runner/work/neovim/neovim/cmake/RunTests.cmake
ninja: build stopped: subcommand failed.
```

File: runtime/doc/develop.txt [hunk 1]
```diff
@@ -173,10 +173,10 @@ Strict "vimdoc" subset:
   (soft-wrapped) instead of preformatted (hard-wrapped) layout common in
   legacy :help docs.
   - Limitation: currently the parser https://github.com/neovim/tree-sitter-vimdoc
-    does not understand numbered listitems, so use a bullet symbol (- or •)
+    does not understand numbered list items, so use a bullet symbol (- or •)
     before numbered items, e.g. "• 1." instead of "1.".
 - Separate blocks (paragraphs) of content by a blank line.
-- Do not use indentation in random places—that prevents the page from using
+- Do not use indentation in random places that prevents the page from using
```
Comment (zeertzjq) [inline runtime/doc/develop.txt:179]:
```suggestion
- Do not use indentation in random places: that prevents the page from using
```

File: runtime/doc/develop.txt [hunk 2]
```diff
@@ -173,10 +173,10 @@ Strict "vimdoc" subset:
   (soft-wrapped) instead of preformatted (hard-wrapped) layout common in
   legacy :help docs.
   - Limitation: currently the parser https://github.com/neovim/tree-sitter-vimdoc
-    does not understand numbered listitems, so use a bullet symbol (- or •)
+    does not understand numbered list items, so use a bullet symbol (- or •)
     before numbered items, e.g. "• 1." instead of "1.".
 - Separate blocks (paragraphs) of content by a blank line.
-- Do not use indentation in random places—that prevents the page from using
+- Do not use indentation in random places: that prevents the page from using
```
Comment (justinmk) [inline runtime/doc/develop.txt:179]: this is an emdash, it's intentional. looks better on web: https://neovim.io/doc/user/develop.html#dev-doc

**Comment by justinmk** [comment]
no thanks

**Pull request closed without merging**